azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventDataBatch.java

```diff
@@ -52,7 +52,13 @@ public final boolean tryAdd(final EventData eventData) {
             throw new IllegalArgumentException("eventData cannot be null");
         }
 
-        final int size = getSize(eventData, events.isEmpty());
+        final int size;
+        try {
+            size = getSize(eventData, events.isEmpty());
+        } catch (java.nio.BufferOverflowException exception) {
```

Comment:
As discussed, let's throw an exception if the size of event data exceeds max-batch size instead of returning false. Otherwise, adding an item will keep failing if the size exceeds max-batch-size. That is, if size of an input is less than max-batch-size and there is no space in the buffer, then return false. If the size of input exceeds the max-batch-size, then throw an exception so that users can fix the issue.

Reply:
done!